### PR TITLE
Lokalize automation trace area

### LINKED ISSUE #18362: Missing in lokalise text from Automation Trace
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Missing in lokalise text from Settings/Automatisations (red marked in screenshot below)

Snag_4038a89

Snag_404e239

Snag_40564a8

### Describe the behavior you expected

Add to lokalise

### What version of Home Assistant Core has the issue?

2023.10.5

### What was the last working version of Home Assistant Core?

not relevant

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Lokalize the messages in the automation trace area.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #18362 #19464
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/trace.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ interface BaseTraceStep {
 }
 
 export interface TriggerTraceStep extends BaseTraceStep {
-  changed_variables: {
```
Comment (silamon) [inline src/data/trace.ts:17]: This is mainly needed to get the gallery pages fixed.

File: src/data/trace.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ interface BaseTraceStep {
...
+  changed_variables?: {
```
Comment (silamon) [inline src/data/trace.ts:17]: This is mainly needed to fix the gallery pages.

File: src/components/trace/hat-trace-timeline.ts [hunk 3]
```diff
@@ -324,20 +329,22 @@ class ActionRenderer {
...
-    )}`,
+      this.hass.localize(
+        "ui.panel.config.automation.trace.messages.triggered_by",
+        {
+          triggeredBy: triggerStep.changed_variables?.trigger.alias
+            ? "alias"
+            : "other",
+          alias: triggerStep.changed_variables?.trigger.alias,
+          triggeredPath: triggerStep.path === "trigger" ? "other" : "trace",
+          trace: this.trace.trigger,
```
Comment (bramkragten) [inline src/components/trace/hat-trace-timeline.ts:340]:
```suggestion
          trigger: this.trace.trigger,
```

File: src/components/trace/hat-trace-timeline.ts [hunk 4]
```diff
@@ -324,20 +329,22 @@ class ActionRenderer {
...
-      this.hass.config
-    )}`,
+      this.hass.localize(
+        "ui.panel.config.automation.trace.messages.triggered_by",
+        {
+          triggeredBy: triggerStep.changed_variables?.trigger.alias
+            ? "alias"
+            : "other",
+          alias: triggerStep.changed_variables?.trigger.alias,
+          triggeredPath: triggerStep.path === "trigger" ? "other" : "trace",
```
Comment (bramkragten) [inline src/components/trace/hat-trace-timeline.ts:339]:
```suggestion
          triggeredPath: triggerStep.path === "trigger" ? "manual" : "trigger",
```

File: src/data/trace.ts (hunk 2 above)
Comment (bramkragten) [reply src/data/trace.ts:17]: Can't we just add an empty `changed_variables` to the mocks in gallery?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**